components/renku_data_services/storage/db.py

```diff
@@ -142,6 +154,95 @@ async def delete_storage(self, storage_id: str, user: base_models.APIUser) -> No
 
             await session.delete(storage[0])
 
+    async def _encrypt_user_secret(self, requested_by: base_models.APIUser, secret_value: str) -> tuple[bytes, bytes]:
+        if requested_by.id is None:
+            raise errors.ValidationError(message="APIUser has no id")
+
+        user_secret_key = await self.user_repo.get_or_create_user_secret_key(requested_by=requested_by)
+
+        return encrypt_user_secret(
+            user_id=requested_by.id,
+            user_secret_key=user_secret_key,
+            secret_service_public_key=self.secret_service_public_key,
+            secret_value=secret_value,
+        )
```

Comment:
issue: how a secret is encrypted is business logic to me and not really part of the db layer (the db layer just stores/loads things, and doesn't really do any transformations except minor ones). I know in the user code I opted for having it in blueprints.py, which is equivalent to the presentation layer, but that's also the wrong place. 

I discussed with Tasko a few weeks back that we should be more consistent with where things go and we'd settled on putting business logic in a `core.py` in the respective modules, though we haven't had a chance to write this up and discuss it with the team.

For secrets specifically though, I wonder if it'd make sense to have a single module or class that deals with all of the secret business logic so it can be reused where needed (the above method is almost identical to the one in users/blueprints.py